internal/page/donor/upload_evidence.go

```diff
@@ -94,18 +99,47 @@ func UploadEvidence(tmpl template.Template, payer Payer, donorStore DonorStore,
 					data.Evidence = lpa.Evidence
 
 				case "pay":
+					var filenames []string
+
+					// TODO decide if we want to delete from S3 now or run a job to delete all infected files
+					lpa.Evidence.Documents = slices.DeleteFunc(lpa.Evidence.Documents, func(d page.Document) bool {
+						if d.VirusDetected {
+							filenames = append(filenames, d.Filename)
+							return true
+						}
+
+						return false
+					})
+
+					if len(filenames) > 0 {
+						if err := donorStore.Put(r.Context(), lpa); err != nil {
+							return err
+						}
+
+						// This is gross. There must be a nicer way to do this...
```

Comment:
create an error type in `validation/error.go` to handle it, we should try to minimise the use of `CustomError`